src/components/diagrams/singleLineDiagram/PositionDiagramPane.js

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright (c) 2022, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+import Dialog from '@mui/material/Dialog';
+import { Alert } from '@mui/material';
+import { useCallback, useEffect, useState } from 'react';
+import { getVoltageLevelSingleLineDiagram } from '../../../utils/rest-api';
+import { useSelector } from 'react-redux';
+import {
+    PARAM_CENTER_LABEL,
+    PARAM_COMPONENT_LIBRARY,
+    PARAM_DIAGONAL_LABEL,
+    PARAM_USE_NAME,
+} from '../../../utils/config-params';
+import PositionDiagram from './PositionDiagram';
+import { AutoSizer } from 'react-virtualized';
+import { useIntl } from 'react-intl';
+import { SLD_DISPLAY_MODE } from '../../network/constants';
+import { SvgType } from './utils';
+
+const PositionDiagramPane = ({
+    open,
+    onClose,
+    voltageLevelId,
+    currentNodeUuid,
+    studyUuid,
+}) => {
+    const useName = useSelector((state) => state[PARAM_USE_NAME]);
+    const centerName = useSelector((state) => state[PARAM_CENTER_LABEL]);
+    const diagonalName = useSelector((state) => state[PARAM_DIAGONAL_LABEL]);
+    const componentLibrary = useSelector(
+        (state) => state[PARAM_COMPONENT_LIBRARY]
+    );
```

Comment:
can we share this ?

Reply:
i can't share this because the hook (useSelector) cannot be called at the top level, must be called in a react function component or a custom hook function

Reply:
yes